detections/endpoint/windows_firewallapi_dll_load_from_temp.yml

```diff
@@ -0,0 +1,64 @@
+name: Windows FirewallAPI DLL Load from Temp
+id: e4cfb996-5fb6-4bcb-93cf-3d42f7cec44c
+version: 1
+date: '2025-08-06'
+author: Teoderick Contreras, Splunk
+status: production
+type: TTP
+description: The following analytic detects suspicious process behavior where firewallapi.dll—a legitimate Windows library responsible for interacting with the Windows Firewall—is loaded from a temporary directory. This is highly atypical, as firewallapi.dll is normally loaded from trusted system locations such as C:\Windows\System32. Loading it from an alternate or user-writable path may indicate DLL search order hijacking, sideloading, or masquerading, techniques commonly used by adversaries to execute code with elevated privileges or bypass security controls. This behavior should be treated as potentially malicious, particularly when observed in conjunction with unsigned binaries, unexpected parent processes, or indicators of privilege escalation and persistence.
+data_source:
+- Sysmon EventID 7
+search: '`sysmon` EventCode=7  ImageLoaded = "*\\temp\\FirewallAPI.dll"  
+  | fillnull 
+  | stats count min(_time) as firstTime max(_time) as lastTime 
+  by Image ImageLoaded dest loaded_file loaded_file_path original_file_name
+  process_exec process_guid process_hash process_id process_name process_path service_dll_signature_exists
```

Comment:
This search is covered by multiple rules leveraging the hijacklibs lookups

`Windows Known Abused DLL Created`
`Windows Known Abused DLL Loaded Suspiciously`
`Windows DLL Search Order Hijacking Hunt with Sysmon`

I suggest you update those searches if needed. No need to create a dedicated rule for just `temp` loading.

Reply:
thanks I didnt see these 2 right away.
Windows Known Abused DLL Loaded Suspiciously
Windows DLL Search Order Hijacking Hunt with Sysmon